ts/Core/Axis/Stacking/StackItem.ts

```diff
@@ -208,86 +212,61 @@ class StackItem {
-                    (boxTop ?
-                        boxTop :
-                        (this.total as any)),
-                0 as any,
-                0 as any,
-                0 as any,
-                1 as any
-            ),
-            yZero = axis.translate(boxBottom ? boxBottom : 0), // stack origin
-            // x position:
-            x = pick(defaultX, chart.xAxis[0].translate(this.x)) +
+            stackBox = this.getStackBox({
                 xOffset,
-            stackBox = defined(y) && this.getStackBox(
-                chart,
-                this,
-                x,
-                y,
-                xWidth,
-                // stack height:
-                Math.abs(y - yZero),
-                axis
-            ),
+                width: xWidth,
+                boxBottom,
+                boxTop,
+                defaultX,
+                xAxis
+            }),
             label = this.label,
-            isNegative = this.isNegative,
-            textAlign = this.textAlign;
+            textAlign = this.textAlign,
+            { align, verticalAlign } = this.alignOptions;
 
         if (label && stackBox) {
             const bBox = label.getBBox(),
                 padding = label.padding;
-
-            let boxOffsetX,
-                isJustify =
+            let isJustify =
                     pick(this.options.overflow, 'justify') === 'justify',
                 visible;
 
-            if (textAlign === 'left') {
-                boxOffsetX = chart.inverted ? -padding : padding;
-            } else if (textAlign === 'right') {
-                boxOffsetX = bBox.width;
-            } else {
-                if (chart.inverted && textAlign === 'center') {
-                    boxOffsetX = bBox.width / 2;
-                } else {
-                    boxOffsetX = chart.inverted ?
-                        (isNegative ? bBox.width + padding : -padding) :
-                        bBox.width / 2;
-                }
-            }
-
-            const boxOffsetY = chart.inverted ?
-                bBox.height / 2 : (isNegative ? -padding : bBox.height);
-
             // Reset alignOptions property after justify #12337
             this.alignOptions.x = pick(this.options.x, 0);
             this.alignOptions.y = pick(this.options.y, 0);
 
-            // Set the stackBox position
-            stackBox.x -= boxOffsetX;
-            stackBox.y -= boxOffsetY;
-
-            // Align the label to the box
-            label.align(this.alignOptions, null as any, stackBox);
+            // Calculate the adjusted Stack position, to take into consideration
+            // The size if the labelBox and vertical alignment as
+            // well as the text alignment. It's need to be done to work with
+            // default SVGLabel.align/justify methods.
+            const { x, y } = this.adjustStackPosition({
+                labelBox: bBox,
+                vAlign: verticalAlign as VerticalAlignValue,
```

Comment:
Can we avoid the `as VerticalAlignValue`? What happens if `verticalAlign` is undefined?

Reply:
It cannot be, because it is set in the constructor. here: https://github.com/highcharts/highcharts/blob/20d15647c198106f74cc014e6ba0982902442e86/ts/Core/Axis/Stacking/StackItem.ts#L105-L114

Reply:
Okay. It's not ideal to override the type checking like that, but it's not the only place we're doing it. I think the appropriate way of doing this would be that `StackItem.alignOptions` were a separate type where the properties were non-optional.

Reply:
seems logical. I will create seperate interface 👍